utils/check-lockfile-links.js

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env node
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const contents = fs.readFileSync(path.resolve(__dirname, '../yarn.lock')).toString();
+
+if (/resolved\s".*(artifactory).*\n/.test(contents)) {
+  console.error(
+    `This 'yarn.lock' file contains resolved urls that are not allowed. This usually happens if you're trying to install npm packages through a proxy like Artifactory. Make sure your npm registry is set to the publicly available one and try running 'yarn install' again.`
```

Comment:
Oops, this part sounds funny:

```npm registry is set to the publicly available one and try running 'yarn install' again.```

let's change it to

```npm registry is set to one that is publicly available and try running 'yarn install' again.```

Reply:
Have you tried using the suggestion feature?

```suggestion
    `This 'yarn.lock' file contains resolved urls that are not allowed. This usually happens if you're trying to install npm packages through a proxy like Artifactory. Make sure your npm registry is set to one that is publicly available and try running 'yarn install' again. `
```

Reply:
I guess that doesn't quite work with a rebase...